src/server/controllers/user.js

```diff
@@ -8,27 +8,41 @@ const jwtSecret = 'mysecret';
 const register = async (req, res) => {
     const { username, password } = req.body;
 
-    const createdUser = null;
+    const hashedPassword = await bcrypt.hash(password, 10)
+
+    const createdUser = await prisma.user.create({
+        data:{
+            username,
+            password,
+
+            username: username,
+            password: hashedPassword
+        }
+    })
 
     res.json({ data: createdUser });
 };
 
 const login = async (req, res) => {
     const { username, password } = req.body;
 
-    const foundUser = null;
+    const foundUser = await prisma.user.findFirst({
+        where: {
+            username
+        }
+    })
 
     if (!foundUser) {
         return res.status(401).json({ error: 'Invalid username or password.' });
     }
 
-    const passwordsMatch = false;
+    const passwordsMatch = await bcrypt.compare (password, foundUser.password)
 
     if (!passwordsMatch) {
         return res.status(401).json({ error: 'Invalid username or password.' });
     }
 
-    const token = null;
```

Comment:
Same thing here, I saw this with someone else's code as well so I think I'm missing the understanding of this code. I had to modify this code to work where you have left it as null here.